derive_miniconf/src/lib.rs

```diff
@@ -174,14 +211,30 @@ fn derive_struct(mut typedef: TypeDefinition, data: syn::DataStruct, atomic: boo
         let field_name = &f.ident;
         quote! {
             #i => {
-                topic.push_str(concat!("/", stringify!(#field_name))).unwrap();
-                if let Some(()) = self.#field_name.recursive_iter(&mut index[1..], topic, value) {
+                let postfix = if topic.len() != 0 {
+                    concat!("/", stringify!(#field_name))
+                } else {
+                    stringify!(#field_name)
+                };
+
+                if topic.push_str(postfix).is_err() {
+                    return None;
```

Comment:
We should probably log/error on this condition, since the user won't be getting all possible keys.

Reply:
Agreed, not sure what we should do here. A panic seems too strong for a user error, I think what you have here is the best we can do, and then document what will happen if TS is too small in `into_iter<TS>()`

Reply:
I've refactored to add a `get_metadata()` function that can be used to retrieve the maximum topic size + recursive depth (length of state vector). This allows us to check the sizes before we even start iteration.